Online Learning Management/Presentation/Controllers/ModulesController.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Online_Learning_Management.Application.DTOs.Module;
+using Online_Learning_Management.Domain.Entities;
+using Online_Learning_Management.Domain.Entities.Module;
+using Online_Learning_Management.Domain.Interfaces;
+using Online_Learning_Management.Infrastructure.Repositories;
+
+namespace Online_Learning_Management.Presentation.Controllers
+{
+    [Route("modules")]
+    [ApiController]
+    public class ModulesController : ControllerBase
+    {
+        private readonly IModuleRepository _moduleRepository;
+        public ModulesController(IModuleRepository moduleRepository)
+        {
+            _moduleRepository = moduleRepository;
+        }
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Module>>> GetAllModules()
+        {
+            var modules = await _moduleRepository.GetAllModulesAsync();
+            return Ok(modules);
+        }
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Module>> GetModulesById(Guid id)
+        {
+            var module = await _moduleRepository.GetModuleByIdAsync(id);
```

Comment:
It would be better to use a ternary to minimize lines of code: `return module == null ? NotFound() : Ok(module);
`